src/main/java/com/marklogic/appdeployer/command/security/DeployRolesCommand.java

```diff
@@ -45,9 +48,22 @@ public void execute(CommandContext context) {
 			logger.info("Redeploying roles that have default permissions and/or references to roles");
 		}
 		removeRolesAndPermissionsDuringDeployment = false;
+		secondPass = true;
 		super.execute(context);
 	}
 
+	/**
+	 * Any file deployed in the first pass, must be a candidate for deployment in the second phase.
+	 * We need to ignore the hashes for these files during the second phase.
+	 */
+	@Override
+	protected void afterResourceSaved(ResourceManager mgr, CommandContext context, File resourceFile, SaveReceipt receipt) {
+		if (!secondPass) {
```

Comment:
I think instead of doing this, can you just do setIncrementalMode(false) in the execute method above before the second super.execute(context)?